### DISCUSSION TITLE
Add float (or finite) type

### DISCUSSION THREAD

**Discussion by davetapley** [mobxjs/mobx-state-tree#1959] [Ideas]
Since we already have an `integer` type:

https://github.com/mobxjs/mobx-state-tree/blob/6a50d79a114811b492b98c37b0bbb35d6c9d9155/packages/mobx-state-tree/src/types/primitives.ts#L113-L117

Do we want one or both of these?
1. A `float` type, which would restrict against `NaN`.
2. A `finite` type, which would restrict against `NaN` and infinities (taking inspiration from [`isFinite`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite)).

I'll concede that both could be implemented with `refinement`,  but so could `integer` and we already have that.

This would have saved me from a bug where I am using `parseFloat` and a `NaN` was sneaking in to my model.
It would be great if I could just specify `float` in my model and have MST throw, instead of having to check for `NaN` manually.

---

Happy to make a PR if it will be accepted 🤞🏻

**Comment by coolsoftwaretyler** [comment]
Hey @davetapley - sorry it took so long for us to get back to you.

We would happily accept a PR to this effect if you're still interested. If you've lost interest in this, I will keep the conversation open and hopefully get this on the roadmap overall + let you know when we release one.
